File: mobile/lib/features/dining/models/menu_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'menu_item.g.dart';

@JsonSerializable()
class MenuItem {
  @JsonKey(name: '_id')
  final String id;
  final String name;
  final double price;
  final String category;
  final String? description;
  final String? imageUrl;
  final List<String>? allergens;
  final bool isActive;

  MenuItem({
    required this.id,
    required this.name,
    required this.price,
    required this.category,
    this.description,
    this.imageUrl,
    this.allergens,
    this.isActive = true,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) =>
      _$MenuItemFromJson(json);

  Map<String, dynamic> toJson() => _$MenuItemToJson(this);
}

File: backend/src/models/MenuItem.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IMenuItem extends Document {
    name: string;
    description?: string;
    price: number;
    category: string;
    imageUrl?: string;
    isActive: boolean;
    allergens: string[];
}

const menuItemSchema = new Schema<IMenuItem>(
    {
        name: { type: String, required: true },
        description: { type: String },
        price: { type: Number, required: true },
        category: { type: String, required: true }, // e.g., 'Appetizers', 'Main Course'
        imageUrl: { type: String },
        isActive: { type: Boolean, default: true },
        allergens: [{ type: String }],
    },
    { timestamps: true }
);

export const MenuItem = mongoose.model<IMenuItem>('MenuItem', menuItemSchema);
